www/js/control/ProfileSettings.jsx

```diff
@@ -0,0 +1,351 @@
+import { Dialog, Button, useTheme } from "react-native-paper";
+import { angularize, getAngularService } from "../angular-react-helper";
+import { useTranslation } from "react-i18next";
+import ExpansionSection from "./ExpandMenu";
+import SettingRow from "./SettingRow";
+import ControlDataTable from "./ControlDataTable";
+import DemographicsSettingRow from "./DemographicsSettingRow";
+import PopOpCode from "./PopOpCode";
+import ReminderTime from "./ReminderTime"
+import useAppConfig from "../useAppConfig";
+
+let controlUpdateCompleteListenerRegistered = false;
+
+//any pure functions can go outside
+const ProfileSettings = () => {
+    // anything that mutates must go in --- depend on props or state... 
+    const { t } = useTranslation();
+    const { appConfig, loading } = useAppConfig();
+    const { colors } = useTheme();
+
+    // get the scope of the general-settings.js file
+    const mainControlEl = document.getElementById('main-control').querySelector('ion-view');
+    const settingsScope = angular.element(mainControlEl).scope();
+    // grab any variables or functions we need from it like this:
+    const { settings, logOut, viewPrivacyPolicy,
+        fixAppStatus, forceSync, openDatePicker,
+        eraseUserData, refreshScreen, endForceSync, checkConsent, 
+        dummyNotification, invalidateCache, showLog, showSensed,
+        parseState, userDataSaved, userData, ui_config } = settingsScope;
+
+    //angular services needed
+    const CarbonDatasetHelper = getAngularService('CarbonDatasetHelper');
+    const UploadHelper = getAngularService('UploadHelper');
+    const EmailHelper = getAngularService('EmailHelper');
+    const ControlCollectionHelper = getAngularService('ControlCollectionHelper');
+    const ControlSyncHelper = getAngularService('ControlSyncHelper');
+    const CalorieCal = getAngularService('CalorieCal');
+    const KVStore = getAngularService('KVStore');
+    const NotificationScheduler = getAngularService('NotificationScheduler');
+
+    if (!controlUpdateCompleteListenerRegistered) {
+        settingsScope.$on('control.update.complete', function() {
+            console.debug("Received control.update.complete event, refreshing screen");
+            refreshScreen();
+            refreshCollectSettings();
+        });
+        controlUpdateCompleteListenerRegistered = true;
+    }
+
+    //functions that come directly from an Angular service
+    const editCollectionConfig = ControlCollectionHelper.editConfig;
+    const editSyncConfig = ControlSyncHelper.editConfig;
+
+    //states and variables used to control/create the settings
+    const [opCodeVis, setOpCodeVis] = useState(false);
+    const [nukeSetVis, setNukeVis] = useState(false);
+    const [carbonDataVis, setCarbonDataVis] = useState(false);
+    const [forceStateVis, setForceStateVis] = useState(false);
+    const [collectSettings, setCollectSettings] = useState({});
+    const [notificationSettings, setNotificationSettings] = useState({});
```

Comment:
Right, and I've actually run into the issues that can be caused by NOT doing it the way react expects in the PR that follows this. I think I wrote a comment about it there but I'm storing an array of Checks that the device is identified to need (moved to one list instead of one per category) and at one point I was mutating the objects in a way that React was not recognizing (because the reference to the check wasn't changing) and so the UI wasn't updating, this is definitely something I personally need to be careful about doing the "react way" so I don't introduce bugs (again) when working with complex states.

Reply:
Exactly! To summarize, we should treat all state as immutable

If we want to update an object which is stored as state, we should make a shallow copy with the change and pass that to the `set` function.
Which is how Abby resolved her issue in the other PR, the `...` operator makes a shallow copy